src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -149,7 +150,10 @@ private static string EscapeExceptBraces(string input)
                 escaped.Append(Regex.Escape(c.ToString()));
             }
         }
- 
-        return escaped.ToString();
+
+        // Here we are not interested in the path itself, only the placeholders.
+        // Path validation is not part of this class, therefore allowing case-insensitive 
+        // matching.
+        return $"^(?i){escaped}";
```

Comment:
@raman-m What we have lost: `/blabla/bliBLI/{PlaceHolder}` wil not match `/blabla/blibli/{PlaceHolder}` because the elements after the host name could be case sensitive, and stricly taken `/blabla/bliBLI` can't match `/blabla/blibli`. But, yes, the behavior with `^(?i)` is the same as the code snippet you provided.